[Kong Fanji] iP

clementkfj opened the pull request:
DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/clementkfj/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

clementkfj committed:
Greet, Echo, Exit

clementkfj committed:
Add, List

clementkfj committed:
Mark as Done

clementkfj committed:
ToDos, Events, Deadlines

clementkfj committed:
A-TextUiTesting

clementkfj committed:
Handle Errors

clementkfj committed:
Delete

clementkfj committed:
Save

clementkfj committed:
Dates and Times

clementkfj committed:
Merge branch 'branch-Level-7'
Merge-Level-7

clementkfj committed:
Fixed merging

clementkfj committed:
Fixed done

clementkfj committed:
Use More OOP

clementkfj committed:
Organize into Packages

clementkfj committed:
Add JUnit Tests

clementkfj committed:
Created a JAR file

clementkfj committed:
JavaDoc

clementkfj committed:
Follow the Coding Standard

clementkfj committed:
Find

clementkfj committed:
Merge branch 'branch-Level-9'
Bring the "Find" command to the program so that users can find tasks based on
keywords.

clementkfj committed:
Merge branch 'branch-A-JavaDoc'
Update the master branch to have javadocs written for most methods and classes.

clementkfj committed:
Fix conflicts in merging CodingStandard

Droffilc13 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import duke.command.*;
```
Perhaps you can consider listing out the commands you are importing and using here?

clementkfj replied:
Thanks! I've updated it! My intellij had a function I did not turn off where it changed imports to * when it reached a certain number!

Droffilc13 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+ */
+public class Parser {
+    /**
+     * This method evaluates a string given to it by the program and produces a Command
```
```suggestion
     * Evaluates a string given to it by the program and produces a Command
```

Droffilc13 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,189 @@
+    }
+
+    /**
+     * A method that deletes a specified task from the local memory using the method deleteLine
```
```suggestion
    * Deletes a specified task from the local memory using the method deleteLine
```

Droffilc13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,209 @@
+    }
+
+    /**
+     * A method that copies the list that is stored in the local file when the program starts running
```
I noticed the same issue in several other places such as Parser class.

Droffilc13 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,169 @@
+     */
+    public void welcomeMessage() {
+        lineProducer();
+        System.out.println("    Hello I'm your friendly task-planning chatbot Duke!" +
```
Perhaps move the ```+``` operator to the beginning of the next line?

Droffilc13 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,169 @@
+    public void welcomeMessage() {
+        lineProducer();
+        System.out.println("    Hello I'm your friendly task-planning chatbot Duke!" +
+                indentationAdder() + "To enter a deadline, please type in this format: 'deadline {title of item} /by d/mm/yyyy hh:mm'" + indentationAdder() +
```
Perhaps you can consider breaking this String up so that it does not exceed 120 characters?

Droffilc13 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,169 @@
+     *
+     * @return A boolean that states false as the program exits.
+     */
+    public boolean byeMessage() {
```
Should ```byeMessage``` be converted to a boolean sounding name?

Droffilc13 commented on test/duke/ParserTest.java:
```diff
@@ -0,0 +1,20 @@
+import duke.command.ExitCommand;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I noticed the same issue in several other places too such as Parser class.

Droffilc13 reviewed (commented):
Overall, logic is easy to follow and classes are neatly organized. 😄 
I noted a few minor coding standard violations.

clementkfj commented:
DukePro
“Your mind is for having ideas, not holding them.” – David Allen (
[source](https://dansilvestre.com/productivity-quotes/)
)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from 
[here](https://github.com/clementkfj/ip)
.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

clementkfj committed:
Merge branch 'add-gradle-support'
Merge gradle support to master branch

clementkfj committed:
Enhance Gradle Testing

clementkfj committed:
Use CheckStyle

clementkfj committed:
Use Checkstyle with branch

clementkfj committed:
Merge branch 'branch-A-CheckStyle'
Merging new Checkstyle branch because I forgot to branch it before.

jaredlhf commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,207 @@
+        Scanner sc = new Scanner(dukeData);
+        int count = 0;
+        boolean changed = false;
+        File temp = new File("data/temp.txt");
```
Perhaps a more descriptive name could be used instead of `temp` ?

jaredlhf commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,207 @@
+    public File doneChanger(String description, String cl, File dukeData) throws IOException {
+        Scanner sc = new Scanner(dukeData);
+        int count = 0;
+        boolean changed = false;
```
Maybe a prefix can be added to the variable name ?
```suggestion
        boolean hasChanged = false;
```

jaredlhf commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,230 @@
+ */
+public class TaskList {
+
+    private ArrayList<Task> xs = new ArrayList<>();
```
Maybe a longer name that is more specific might be better for readability?

jaredlhf commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,51 @@
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        boolean allDone;
```
Can maybe be made to sound more like a boolean ?

jaredlhf commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,230 @@
+     * @param dukeData The file that resides in the user's local memory.
+     * @throws DukeException An exception thrown as Scanner requires a FileNotFoundException to be handled.
+     */
+    public void fileCopy(File dukeData) throws FileNotFoundException {
```
Maybe`copyFile` might be a better name ?

jaredlhf reviewed (commented):
Minor naming mistakes. Overall, code is readable and names are mostly appropriate.

clementkfj committed:
Implement GUI

clementkfj committed:
Merge branch 'branch-Level-10'
Merging branch-Level-10 into the main branch

clementkfj committed:
Add FXML

clementkfj committed:
Add assertions to code

clementkfj committed:
Add assertions to code with message

Assertions help to keep track of logically impossible situations, by
checking if the methods accidentally allow something illogical to bypass it.

Used assert to after some logical statements that should not produce any errors
so as to ensure that the code does not give birth to impossible outcomes.

This is to ensure that the when a soemthing thought to be impossible to produce
actually happens, the program can actually detect it and helps in debugging.

clementkfj committed:
Merge branch 'branch-A-Assertions'
Merge branch-A-Assertions to master

Merging to master branch allows the master branch to have the assertions
that were added from the branch to be used.

Just merging the branch to the main branch.

So as to keep track of an upstream branch such that we can have it reflected
on the 2103T website.

clementkfj committed:
Improve Code Quality

Code now is messy and sometimes hard to decipher and read.

In order to debug easier in the future, code quality is required to
improve so that others that view the code will also find it more readable.

Improved some lines of code to adhere to guard clauses as well as naming
conventions(Although not all as some of them are a little difficult to change).

Changing naming conventions will allow users to understand the codes and their functions
much easier and guard clauses make the code much more readable.

clementkfj committed:
Merge branch 'branch-A-CodeQuality'
Merging branch A CodeQuality to master branch

Currently both branches are separated and only the branch-A-CodeQuality
has improved readable code.

People will usually view the master branch instead of the branch so there
is a need to merge both branches.

Merging both branches in this commit.

To improve readability.

clementkfj committed:
Add BCD-Extension, C-FriendlierSyntax

Currently, code takes in specific commands like delete, todo, deadline, etc.
These commands can be long to type out and thus not feasible.

The commands are long and requires a lot of effort just by typing in.
As such, we need to shorten commands with same functionality in order
to improve user experience.

The code followed C-FriendlierSyntax guidelines and shortened commands in order
for users to have the option to use shortcuts. E.g. deadline can be shortened to
d.

Improves user experience, for example, only needing to type l instead of list to access
their list of tasks.

Welcome message gives a brief rundown of what shortcuts there are, help command also
gives another list of commands that are useful.

clementkfj committed:
Merge branch 'branch-BCD-Extension'
Merge BCD-Extension branch with master

BCD-Extension branch consists of new features such as C-FriendlierSyntax
that improves user command typing experience.

clementkfj committed:
Additional comments for Merging PR

clementkfj committed:
Merge pull request #1 from clementkfj/branch-Merging-PR

Additional comments for Merging PR

clementkfj committed:
Fix Parser and improve code quality

Parser currently is buggy with the new introduction of friendlier syntax.

Code needs to be modified so user input will not cause any bugs and will
be accurate.

Abstracted methods out of main parse method and changed methods that used
substring to split.

This improves code readability as well as cause it to be less buggy as user
input is not used separated by a fixed number but by splitting by words.

This commit helps improve SLAP in this program as well as arrow headed style of coding.

clementkfj committed:
Final commit with style changes

clementkfj committed:
Add stage name

clementkfj committed:
Update user guide

clementkfj committed:
Update main README

Teach users what Duke currently does as well as how to run Duke.

clementkfj committed:
Merge pull request #3 from clementkfj/clementkfj-Readme-update

Update main README

clementkfj committed:
Update README.md
